src/store.js

```diff
@@ -1,94 +1,180 @@
 import Vue from 'vue'
-import { $set } from './helpers'
-import { loadState, dumpState } from './storage'
-import { subscribe } from './watcher'
+import { namespaceSeparator, getModuleByNamespace } from './namespace'
+import { registerStorage } from './storage'
+import { setWatcher, subscribe } from './watcher'
+import { callIfFunction } from './utils'
 
 export function createStore ({
-  ctx,
   state,
+  hydrate,
   actions,
   getters,
-  hydrate,
-  localStorage,
-  sessionStorage
-}) {
-  const initialState = state(ctx)
+  modules,
+  storage
+}, vm) {
```

Comment:
vm can be passed as second arg. Providing an existing Vue instance if a small performance improvement because calling `new Vue()` in watcher.js has a measurable performance cost